### ISSUE TITLE
Question about DNN or D

### ISSUE THREAD

**Issue by cloverzyy** [golmschenk/sr-gan#9]
In age/srgan.py, we can choose to set model_architecture. So in model_setup, does it mean that, if dcgan is selected, D and DNN are both generative adversus-network models like DcGAN? Conversely, if vgg is selected, both D and DNN are models of VGG？
`model_architecture = 'dcgan'  # dcgan or vgg
  def model_setup(self):
      """Prepares all the model architectures required for the application."""
      if model_architecture == 'vgg':
          self.G = Generator(image_size=224)
          self.D = vgg16(pretrained=True, num_classes=1)
          self.DNN = vgg16(pretrained=True, num_classes=1)
      else:
          self.G = Generator()
          self.D = Discriminator()
          self.DNN = Discriminator()
`

According to the code of age part and the paper, I drew a simple loss function calculation flow chart. Does this mean that the three red circles in the figure below: DNN and the two D's are actually the same module?
<img width="666" alt="image" src="https://user-images.githubusercontent.com/33452748/216875006-0224731a-cc8a-4768-8d92-40cc16e35d78.png">

**Comment by golmschenk** [comment]
Almost. DNN and D have the same neural network architecture but are trained separately. The idea is DNN is trained with a standard (non-GAN) approach, and the D is trained with the SR-GAN approach (which also includes the labeled loss, along with the other losses). DNN doesn't need to be present to train D, and D doesn't need to be present to train DNN. The reason they both exist in the same code is purely to compare what happens when you train via both methods. That being, to demonstrate that the same network can be trained with less labeled data to produce a better result. DNN has only the standard (non-GAN) labeled loss. D is trained using the standard labeled loss, but we add on the two additional GAN-related losses (getting the correct features on the unlabeled data and getting contrasting features on the fake data). In that way, D has access to all three losses, and DNN has just the standard labeled loss. D and DNN are the same network but just trained in different ways. And D is a part of the larger GAN structure used to train D, but then during inference time, D is used by itself. Finally, yes, changing `model_architecture` will change both the D and DNN together (because we want to compare how well the two methods work compared against one another). Hopefully, that clears it up!
